File: cc/zco/bamboo-art/sol.cpp
#include <bits/stdc++.h>
using namespace std;

const int maxlen = 1e5 + 3;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int n; cin >> n;
    int a[n];
    for (int i = 0; i < n; i++) cin >> a[i];
    sort(a, a+n);

    int dp[n][maxlen] = {};
    int ans = 0;
    for (int i = 1; i < n; i++) {
        for (int j = 0; j < i; j++) {
            int d = a[i] - a[j];
            dp[i][d] = dp[j][d] + 1;
            ans = max(dp[i][d], ans);
        }
    }

    printf("%d\n", ans+1);
}

File: cses/searching/1141/sol.py
n = int(input())
a = [int(x) for x in input().split()]

s = set()
l = 0
r = 0
ans = 0

while r < n:
    while a[r] in s:
        s.remove(a[l])
        l += 1
    s.add(a[r])
    ans = max(ans, r - l + 1)
    r += 1

print(ans)
